Trimble.Modus.Components/Controls/TMInput/TMInput.cs

```diff
@@ -0,0 +1,871 @@
+            _borderlessEntry.SetBinding(Entry.KeyboardProperty, new Binding(nameof(Keyboard), BindingMode.TwoWay, source: this));
+            _borderlessEntry.SetBinding(Entry.PlaceholderProperty, new Binding(nameof(Placeholder), BindingMode.TwoWay, source: this));
+            _borderlessEntry.SetBinding(Entry.ReturnCommandProperty, new Binding(nameof(ReturnCommand), BindingMode.TwoWay, source: this));
+            _borderlessEntry.SetBinding(Entry.ReturnCommandParameterProperty, new Binding(nameof(ReturnCommandParameter), BindingMode.TwoWay, source: this));
+            _borderlessEntry.SetBinding(Entry.FontAttributesProperty, new Binding(nameof(FontAttributes), BindingMode.TwoWay, source: this));
+            _borderlessEntry.SetBinding(Entry.FontFamilyProperty, new Binding(nameof(FontFamily), BindingMode.TwoWay, source: this));
+            _borderlessEntry.SetBinding(Entry.FontSizeProperty, new Binding(nameof(FontSize), BindingMode.TwoWay, source: this));
+            _borderlessEntry.SetBinding(Entry.ReturnTypeProperty, new Binding(nameof(ReturnType), BindingMode.TwoWay, source: this));
+            _borderlessEntry.SetBinding(Entry.PlaceholderColorProperty, new Binding(nameof(PlaceholderColor), BindingMode.TwoWay, source: this));
+            _borderlessEntry.SetBinding(Entry.MaxLengthProperty, new Binding(nameof(MaxLength), BindingMode.TwoWay, source: this));
+
+            _titleLabel.SetBinding(Label.TextProperty, new Binding(nameof(TitleText), BindingMode.TwoWay, source: this));
+
+            _leftIcon.SetBinding(ImageButton.CommandProperty, new Binding(nameof(LeftIconCommand), BindingMode.TwoWay, source: this));
+            _leftIcon.SetBinding(ImageButton.CommandParameterProperty, new Binding(nameof(LeftIconCommandParameter), BindingMode.TwoWay, source: this));
+            _rightIcon.SetBinding(ImageButton.CommandProperty, new Binding(nameof(RightIconCommand), BindingMode.TwoWay, source: this));
+            _rightIcon.SetBinding(ImageButton.CommandParameterProperty, new Binding(nameof(RightIconCommandParameter), BindingMode.TwoWay, source: this));
+
+            _helperLabel.SetBinding(Label.TextProperty, new Binding(nameof(HelperText), source: this));
+        }
+
+        /// <summary>
+        /// Listens to entry focus change and updates the border style
+        /// </summary>
+        /// <param name="sender">object</param>
+        /// <param name="e"></param>
+        private void OnEntryFocusChanged(object sender, FocusEventArgs e)
+        {
+            if (sender is Entry entry && entry.IsFocused)
+            {
+                _border.Stroke = (Color)BaseComponent.colorsDictionary()["TrimbleBlue"];
+                _border.StrokeThickness = 2;
+                Focused?.Invoke(this, e);
+            }
+            else
+            {
+                _border.Stroke = (Color)BaseComponent.colorsDictionary()["Black"];
+                _border.StrokeThickness = 1;
+                Unfocused?.Invoke(this, e);
```

Comment:
When unfocused, the error/success message should go away and helper text should be displayed

Reply:
Done